### ISSUE TITLE
Build fails: CARGO_CFG_TARGET_FEATURE env-var is not defined or is not UTF-8: NotPresent (architecture: armv7)

### ISSUE THREAD

**Issue by yurivict** [qdrant/quantization#23]
See [log](https://pkg-status.freebsd.org/ampere3/data/132releng-armv7-default/7036c2716ded/logs/qdrant-1.7.3_3.log).

Project: https://qdrant.tech

rust-1.75.0
FreeBSD 14.0

**Comment by KastanDay** [comment]
I faced this issue when building on `ppc64le` machine (IBM Power9). 

Build: `cargo build --release --bin qdrant`

Error: 
```
error: failed to run custom build command for `quantization v0.1.0 (https://github.com/qdrant/quantization.git#0caf67d9)`
thread 'main' panicked at /home/kastanday/.cargo/git/checkouts/quantization-e4ab6fc511580078/0caf67d/quantization/build.rs:16:10:
CARGO_CFG_TARGET_FEATURE env-var is not defined or is not UTF-8: NotPresent
```

**Comment by timvisee** [comment]
Are you sure your Rust/cargo installation is up-to-date?

It seems to break on a missing `CARGO_CFG_TARGET_FEATURE` variable. But if compiling with cargo, it should always be there according to [this](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts).

**Comment by workingjubilee** [comment]
I do not believe that is correct. I believe it is there when there is a value to set there, and if there is none then there is no information to be provided in that regard.
